pkg/controller/migplan/gvk.go

```diff
@@ -0,0 +1,414 @@
+	srcCluster, err := r.Plan.GetSourceCluster(c)
+	if err != nil {
+		return errors.Wrap(err, "Error reading srcMigCluster")
+	}
+
+	client, err := r.getClient(c, srcCluster)
+	if err != nil {
+		return errors.Wrap(err, "Can't compile dynamic client for srcCluster")
+	}
+
+	r.SrcClient = client
+
+	return nil
+}
+
+func (r *Compare) getDiscovery(c client.Client, cluster *migapi.MigCluster) (*discovery.DiscoveryClient, error) {
+	config, err := cluster.BuildRestConfig(c)
+	if err != nil {
+		return nil, errors.Wrap(err, "Can't get REST config from a cluster")
+	}
+
+	return discovery.NewDiscoveryClientForConfig(config)
+}
+
+func (r *Compare) getClient(c client.Client, cluster *migapi.MigCluster) (dynamic.Interface, error) {
+	config, err := cluster.BuildRestConfig(c)
+	if err != nil {
+		return nil, errors.Wrap(err, "Can't get REST config from a cluster")
+	}
+
+	return dynamic.NewForConfig(config)
+}
+
+func (r *Compare) collectUnsupportedMapping(unsupportedResources []schema.GroupVersionResource) (map[string][]schema.GroupVersionResource, error) {
+	unsupportedNamespaces := map[string][]schema.GroupVersionResource{}
+	for _, gvr := range unsupportedResources {
+		namespaceOccurence, err := r.occurIn(gvr)
+		if err != nil {
+			return nil, errors.Wrapf(err, "Unable to collect namespace occurences for GVR: %s", gvr)
+		}
+
+		for _, namespace := range namespaceOccurence {
+			if inNamespaces(namespace, r.Plan.GetSourceNamespaces()) {
+				_, exist := unsupportedNamespaces[namespace]
+				if exist {
+					unsupportedNamespaces[namespace] = append(unsupportedNamespaces[namespace], gvr)
+				} else {
+					unsupportedNamespaces[namespace] = []schema.GroupVersionResource{gvr}
+				}
+			}
+		}
+	}
+
+	return unsupportedNamespaces, nil
+}
+
+func (r *Compare) occurIn(gvr schema.GroupVersionResource) ([]string, error) {
+	namespacesOccured := []string{}
+	options := metav1.ListOptions{}
+	resourceList, err := r.SrcClient.Resource(gvr).List(options)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Error while listing: %s", gvr)
+	}
+
+	for _, res := range resourceList.Items {
+		if !inNamespaces(res.GetNamespace(), namespacesOccured) {
+			namespacesOccured = append(namespacesOccured, res.GetNamespace())
+		}
+	}
+
+	return namespacesOccured, nil
+}
+
+func collectResources(discovery discovery.DiscoveryInterface) ([]*metav1.APIResourceList, error) {
+	resources, err := discovery.ServerResources()
+	if err != nil {
+		return nil, errors.Wrap(err, "Unable to get a list of resources for a GroupVersion on srcCluster")
+	}
+
+	for _, res := range resources {
+		res.APIResources = namespaced(res.APIResources)
+		res.APIResources = excludeSubresources(res.APIResources)
+		// Some resources appear not to have permissions to list, need to exclude those.
+		res.APIResources = listAllowed(res.APIResources)
+	}
+
+	return resources, nil
+}
+
+func (r *Compare) unsupportedResources(resourceDiff []*metav1.APIResourceList) ([]schema.GroupVersionResource, error) {
```

Comment:
There is no need here for a function type.  (r *Compare) is not used.

Reply:
Fixed, thanks :)